### PR TITLE
Implement getProcessInstances for ES

### LINKED ISSUE #24930: Add ES incident update repository implementation
# Description

Implement all the methods in `IncidentUpdateRepository` for Elasticsearch. This should include integration tests of each method against an actual ES instance itself.

I would recommend splitting this into multiple PR, one per method. Inherit from the `NoopIncidentUpdateRepository` initially, and then go from there. This can be easily parallelized among multiple people.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

Implements `getProcessInstances` for the `ElasticsearchIncidentUpdateRepository`. One difference from the original implementation here, is that I also check the join relation property - the original only did for the FNIs, but I figured it's risky to rely on the ID always being correct.

> [!Note]
> With the additional tests to the clear scroll stuff, and the changes to fix it, I'm now thinking a good follow up would be to extract that to a utility you can test - that way, we don't need to keep adding new tests for each scrolling method using it. The fact it scrolls is pretty deep implementation detail imo.

## Related issues

closes #24930

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
 import java.util.stream.Collectors;
 import org.slf4j.Logger;
 
-public final class ElasticsearchIncidentUpdateRepository extends NoopIncidentUpdateRepository {
+public final class ElasticsearchIncidentUpdateRepository implements IncidentUpdateRepository {
```
Comment (entangled90) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:56]: :tada:

**Comment by npepinpe** [comment]
Did my force push discard your review? I just rebased on the other branch, nothing else :sweat_smile:

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java [hunk 2]
```diff
@@ -142,6 +142,29 @@ public CompletionStage<Collection<Document>> getFlowNodeInstances(
...
+            .index(listViewAlias)
+            .source(s -> s.filter(f -> f.includes(ListViewTemplate.TREE_PATH)))
+            .query(q -> q.bool(b -> b.must(idQ, typeQ)));
+
+    return fetchUnboundedDocumentCollection(
+        request,
+        ProcessInstanceForListViewEntity.class,
+        hit ->
+            new ProcessInstanceDocument(
+                hit.id(), hit.index(), Long.parseLong(hit.id()), hit.source().getTreePath()));
```
Comment (entangled90) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:165]: :thought_balloon:  do we need to handle potential errors when applying the transformer? 
 `Long.parseLong` may throw an exception in the transformer.

We have three options:
- throw an exception: the future may never be completed and the exception is bubbled up
- we fail the future
- we may also want to skip that document, log the exception and keep processing the rest of the documents 

I don't have enough context to know what's the better approach here. 
We don't necessarily need to discuss it in this PR.

Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:165]: I believe we do handle the future when the exception is thrown, but assuming it's deterministic, we would get stuck there.

As this was the previous behavior, I would assume this is ok. That said, it's on my improvement list to not use the ID but instead the key from the document (which is already a long).

**Comment by npepinpe** [comment]
Is there anything you'd like to discuss before approval? :thinking:

**Review by entangled90 — APPROVED** [review]

**Pull request merged**